src/state/modals/index.tsx

```diff
@@ -238,18 +245,22 @@ export function Provider({children}: React.PropsWithChildren<{}>) {
     [setIsModalActive, setActiveModals],
   )
 
-  unstable__openModal = openModal
-
   const closeModal = React.useCallback(() => {
     let totalActiveModals = 0
+    let wasActive = false
     setActiveModals(activeModals => {
+      wasActive = activeModals.length > 0
       activeModals = activeModals.slice(0, -1)
       totalActiveModals = activeModals.length
       return activeModals
     })
     setIsModalActive(totalActiveModals > 0)
+    return wasActive
```

Comment:
Same, this kind of pattern is not to be trusted — there's no guarantee the function above would have executed by now

Reply:
Fixed